awx/main/tasks/jobs.py

```diff
@@ -105,14 +103,12 @@ class BaseTask(object):
     model = None
     event_model = None
     abstract = True
+    callback_class = RunnerCallback
 
     def __init__(self):
+        self.runner_callback = self.callback_class()
         self.cleanup_paths = []
-        self.parent_workflow_job_id = None
-        self.host_map = {}
-        self.guid = GuidMiddleware.get_guid()
-        self.job_created = None
-        self.recent_event_timings = deque(maxlen=settings.MAX_WEBSOCKET_EVENT_RATE)
+        self.runner_callback.event_data_key = self.event_data_key
```

Comment:
As I search this file for `event_data_key`, I don't see any reason the task class should own it. I feel that it should be moved entirely to the callback class. So, I picture:

```python
class RunnerCallbackForProjectUpdate(RunnerCallback):
    event_data_key = 'project_update_id'
```

That means that you would have to introduce a new class for every job type that has an event_data_key defined. To me, that feels like the right solution. That will clean up a lot of the work in `__init__` methods here.